Istio migration docs (#4246)
kyma-project/kyma: docs/application-connector/02-01-application-connector.md

![Architecture Diagram](./assets/001-application-connector.svg)

## Istio Ingress Gateway

> The Application Connector is being exposed by Istio Ingress Gateway as the rest of the Kyma.

Reviewer: ```suggestion
The Istio Ingress Gateway exposes the Application Connector and other Kyma components.
```

The DNS name of the Ingress is cluster-dependant and follows the `gateway.{cluster-dns}` format, for example `gateway.servicemanager.cluster.kyma.cx`.
Istio Ingress Gateway secures the endpoints with certificate validation. Each call must include a valid client certificate.

> You can access every exposed Application (App) by using the assigned path. For example, to reach the gateway for the `user-custom` App, use this URL: `gateway.servicemanager.cluster.kyma.cx/user-custom`. 

Reviewer: ```suggestion
You can access every exposed Application (App) using the assigned path. For example, to reach the gateway for the `user-custom` App, use `gateway.servicemanager.cluster.kyma.cx/user-custom`. 
```

## Application Connectivity Validator

> The Application Connectivity Validator is responsible for validating subject of the client certificate. It is deployed per Application and it proxies requests to the Application Registry and the Events Service.

Reviewer: ```suggestion
The Application Connectivity Validator verifies the subject of the client certificate. It is deployed per Application and it proxies requests to the Application Registry and the Events Service.
```
